spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/configuration/ZeebeClientConfigurationSpringImpl.java

```diff
@@ -175,43 +277,86 @@ public ScheduledExecutorService jobWorkerExecutor() {
 
   @Override
   public boolean ownsJobWorkerExecutor() {
-    return properties.ownsJobWorkerExecutor();
+    return getOrLegacyOrDefault(
+        "ownsJobWorkerExecutor",
+        zeebeClientExecutorService::isOwnedByZeebeClient,
+        properties::ownsJobWorkerExecutor,
+        DEFAULT.ownsJobWorkerExecutor(),
+        configCache);
   }
 
   @Override
   public boolean getDefaultJobWorkerStreamEnabled() {
-    return properties.getDefaultJobWorkerStreamEnabled();
+    return getOrLegacyOrDefault(
+        "DefaultJobWorkerStreamEnabled",
+        () -> camundaClientProperties.getZeebe().getDefaults().getStreamEnabled(),
+        properties::getDefaultJobWorkerStreamEnabled,
+        DEFAULT.getDefaultJobWorkerStreamEnabled(),
+        configCache);
   }
 
   @Override
   public boolean useDefaultRetryPolicy() {
-    return properties.useDefaultRetryPolicy();
+    return false;
   }
 
   @Override
   public boolean preferRestOverGrpc() {
-    return false;
+    return camundaClientProperties.getZeebe().isPreferRestOverGrpc();
   }
 
-  private CredentialsProvider initCredentialsProvider() {
+  private CredentialsProvider identityCredentialsProvider() {
+    if (!(authentication instanceof DefaultNoopAuthentication)) {
+      return new IdentityCredentialsProvider(authentication);
+    }
+    return null;
+  }
 
-    if (commonConfigurationProperties.getEnabled()
-        && !(authentication instanceof DefaultNoopAuthentication)) {
-      return new CredentialsProvider() {
-        @Override
-        public void applyCredentials(final CredentialsApplier applier) {
-          final Map.Entry<String, String> authHeader = authentication.getTokenHeader(Product.ZEEBE);
-          applier.put(authHeader.getKey(), authHeader.getValue());
-        }
+  private String composeGatewayAddress() {
```

Comment:
🔧 

```suggestion
  private String composeGatewayAddress() {
    ZeebeGatewayUrl gatewayUrl = camundaClientProperties.getZeebe().getGatewayUrl();
    int port = gatewayUrl.getPort();
    int defaultPort = gatewayUrl.getDefaultPort();
    String host = gatewayUrl.getHost();

    if (port != -1) {
        return composeAddressWithPort(host, port, "Gateway port is set");
    } else if (defaultPort != -1) {
        return composeAddressWithPort(host, defaultPort, "Gateway port has default");
    } else {
        LOG.debug("Gateway cannot be determined, address will be '{}'", host);
        return host;
    }
}

private String composeAddressWithPort(String host, int port, String debugMessage) {
    String gatewayAddress = host + ":" + port;
    LOG.debug(debugMessage + ", address will be '{}'", gatewayAddress);
    return gatewayAddress;
}
```

Reply:
Applied these changes in my latest commit, thanks 👍